group-coordinator/src/main/java/org/apache/kafka/coordinator/group/taskassignor/ProcessState.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group.taskassignor;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Collections.unmodifiableSet;
+import static org.apache.kafka.common.utils.Utils.union;
+
+public class ProcessState {
```

Comment:
Do we actually need this data structure?
Wouldn't it be enough to have a map from process ID to member (`Owner` in this PR) and maintain the assigned tasks in the member?
At the moment, you keep the assigned tasks in two places -- here and in the maps `activeTasksAssignments` and `standbyTasksAssignments` in the assignor. Is this really needed?

Reply:
Yes, you are right. Having assigned tasks in two places is not an optimal solution.
By removing `ProcessState` and adding a map from `Process ID` to `member`, we need the `member` data structure (the map must be `PId` -> `Set<Member>`). I think keeping `ProcessState` and removing `activeTasksAssignments` and `standbyTasksAssignments` is a better idea since `ProcessState` methods help keep the code cleaner/more readable.